### PR TITLE
Parsed satoshi core txvalidjson tests

### PR DESCRIPTION
**Pull request by tphyahoo**
use tx_valid.json from bitcoin core, parse it for same style of transaction tests that were already running.

65 / 83 tests fail. 

I haven't done an assessment of all the various fail cases, and probably some are spurious. Hope to do more on that soon.

I'm confident that there are (at least) csv/cltv related fails here, that can be used as a starting point for #214.  So I think it's worth merging for that alone.

### CODE REVIEW

**Comment by tphyahoo** [comment]
Is it permitted to merge failing tests to master? 

The easiest way to get all passing is to revert the changes to Main.hs in test suite (which just skips the tests for now).

**Comment by jprupp** [comment]
Skip the tests that are failing and add TODOs to re-enable when fixed. The test suite must always pass.

File: haskoin-core/Network/Haskoin/Script/Types.hs [hunk 1]
```diff
@@ -53,7 +53,7 @@ instance Binary Script where
             empty <- isEmpty
             if empty
                 then return []
-                else liftM2 (:) get getScriptOps
+                else (pure (:)) <*> get <*> getScriptOps
```
Comment (jprupp) [inline haskoin-core/Network/Haskoin/Script/Types.hs]: I think this is the same as:

``` haskell
(:) <$> get <*> getScriptOps
```

Comment (tphyahoo) [inline haskoin-core/Network/Haskoin/Script/Types.hs]: yes, it's precisely the same.

**Comment by tphyahoo** [comment]
Test suite now passes, with

```
+    passingTests = [0..5] ++ [8] ++ [11..13] ++ [16..18] ++ [20] ++ [52]
+    isCurrentlyPassing (_, testNum) = elem testNum passingTests
```

15 new passing tests.

File: haskoin-core/Network/Haskoin/Script/Types.hs (hunk 1 above)
Comment (jprupp) [inline haskoin-core/Network/Haskoin/Script/Types.hs]: Can you change it for the simpler version using `(<$>)`? I will merge when you do and @plaprade gives his ack. Consider this pull request "acked" by me.

Comment (tphyahoo) [inline haskoin-core/Network/Haskoin/Script/Types.hs]: ok I made the change: fcbf97b

**Comment by tphyahoo** [comment]
I like pure f <*> instead of f <$> (which are identical) because it seems more symmetrical to me, and the translation from pure

```
f a b c
```

to 

```
pure f <*> pure a <*> pure b <*> pure c
```

rather than

```
f <$> pure a <*> pure b <*> pure c
```

seems more symmetrical to me.

But, the <$> variant sees to be more prevalent in the community.

Of course, it doesn't really matter since both are totally equivalent.

Just 2c.

**Comment by jprupp** [comment]
I have merged this as a single squashed commit. GitHub does not notice that the code has been merged.

**Pull request closed without merging**